src/v/cloud_storage/inventory/ntp_hashes.cc

```diff
@@ -45,6 +45,16 @@ ntp_path_hashes::ntp_path_hashes(
   , _rtc{_as}
   , _ctxlog{cst_log, _rtc, _ntp.path()} {}
 
+ntp_path_hashes::ntp_path_hashes(ntp_path_hashes&& other) noexcept
```

Comment:
nit: could it be implemented as `default`?

Reply:
I actually tried to default first but ran into issues with the `_rtc` as it looks like it can not be moved, so had to build a new rtc using the `other` abort source. This implementation is to work around the retry chain node being constructed and everything else moves.